pkg/ipcache/ipcache.go

```diff
@@ -417,9 +417,10 @@ func (ipc *IPCache) UpsertMetadata(prefix netip.Prefix, src source.Source, resou
 	ipc.TriggerLabelInjection()
 }
 
-func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix, resource ipcacheTypes.ResourceID, aux ...IPMetadata) {
+// RemoveMetadata removes the metadata of a given IP from the ipcache metadata
+func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix) {
 	ipc.metadata.Lock()
-	ipc.metadata.remove(prefix, resource, aux...)
+	ipc.metadata.removeAllForPrefix(prefix)
 	ipc.metadata.Unlock()
```

Comment:
This will actually break the intended behavior of the metadata resolution. Essentially, there should be an equal number of upserts to removals. In other words, for every upsert-metadata, there is a remove-metadata. This breaks that assumption, which will remove metadata from a prefix prematurely.